mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/MimePart.java

```diff
@@ -303,7 +303,8 @@ private boolean isHtmlSubType(MimePart mimePart) {
     @JsonIgnore
     public Stream<MimePart> getAttachmentsStream() {
         return attachments.stream()
-                .flatMap((mimePart) -> Stream.concat(Stream.of(mimePart), mimePart.getAttachmentsStream()));
+                .flatMap(mimePart -> Stream.concat(Stream.of(mimePart), mimePart.getAttachmentsStream()))
+                .filter(mimePart -> mimePart.contentDisposition.isPresent());
```

Comment:
Not sure to understand fully this, can you explain?

Reply:
So far we considered all mime parts as attachments, even 'main' body parts, and multiparts.

That's not what an attachment is.

An attachment is a mimepart, but it needs a `Content-Disposition` header, typically with a value `inline` or `attachment`. Without it, it is just NOT a mime part, indexing that mime part is both functionnally and performance wise plenty wrong.

Reply:
Thanks for the explanation